### PR TITLE
Fix clang-tidy error in Keeper `Changelog`

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Coordination/Changelog.cpp [hunk 1]
```diff
@@ -157,21 +157,24 @@ class ChangelogWriter
...
-        if (log_is_complete)
-            rotate(record.header.index);
-
-        // writing at least 1 log is requirement - we don't want empty log files
-        // we use count() that can be unreliable for more complex WriteBuffers, so we should be careful if we change the type of it in the future
-        const bool log_too_big = record.header.index != getStartIndex() && log_file_settings.max_size != 0
-            && initial_file_size + file_buffer->count() > log_file_settings.max_size;
-
-        if (log_too_big)
+        if (const bool log_is_complete = record.header.index - getStartIndex() == current_file_description->expectedEntriesCountInLog();
```
Comment (CheSema) [inline src/Coordination/Changelog.cpp:160]: It is hard to parse. Why don't just write like this?
`if (record.header.index - getStartIndex() == current_file_description->expectedEntriesCountInLog())`
We could introduce a function with explanation name for this expression.

Comment (antonio2368) [reply src/Coordination/Changelog.cpp]: This was autopilot change, I saw a variable that is used for if only, and I just turned it into `if-with-initializer` without noticing that the variable is a bool and can be written as you said.
I changed to your suggestion + added a comment to keep the clarity. Thanks for noticing!

**Review by CheSema — APPROVED** [review]

**Comment by yakov-olkhovskiy** [comment]
I'm afraid this didn't work?
https://s3.amazonaws.com/clickhouse-builds/45375/5b3054492ddad332393a73db547c301633afc1a9/binary_tidy/build_log.log
PR #45375 and couple others...

**Comment by antonio2368** [comment]
@yakov-olkhovskiy clang-tidy build was green here and now in the MasterCI so I would say it's okay.
